src/openforms/logging/tests/test_events.py

```diff
@@ -33,3 +38,42 @@ def test_submission_start(self):
 
             log = TimelineLogProxy.objects.last()
             self.assertEqual(_("Anonymous user"), log.fmt_user)
+
+    def test_submission_logic_evaluated(self):
+        submission = SubmissionFactory.create()
+        form = FormFactory.create()
+        form_step = FormStepFactory.create(form = form)
+        submission_step = SubmissionStepFactory.create(submission = submission, form_step = form_step)
+        json_logic_trigger = {
+            ">": [
+                {
+                "date": {
+                    "var": "birthdate"
+                }
+                },
+                {
+                "date": "2022-06-20"
+                }
+            ]
+        }
+
+        actions = [{
+            "component": "firstname",
+            "formStep": "",
+            "action": {
+                "type": "property",
+                "property": {
+                "value": "disabled",
+                "type": "bool"
+                },
+                "state": True
+            }
+        }]
+        rule = FormLogic(form=form, json_logic_trigger=json_logic_trigger, actions=actions)
+        logevent.submission_logic_evaluated(submission, rule)
+        log = TimelineLogProxy.objects.filter(
+            template="logging/events/submission_logic_evaluated.txt", 
+            object_id = submission.id)[0]
+        
+        self.assertIsNotNone(log)
```

Comment:
1. Accessing by index `[0]` does not guarantee there haven't been multiple records created
2. `queryset[0]` will never be None, as that would raise an `IndexError` instead.

-> assert the expected number of objects with `qs.count()` and/or use `queryset.get()` to guarantee only one record exists.

Then, make assertions on the content of the object, here you'd want to validate `expected_data` and validate that the introspection from formio was done correctly (i.e. correct step name, correct component label... are recorded).